main/server/init.js

```diff
@@ -0,0 +1,42 @@
+import { fork } from 'child_process'
+import { join } from 'path'
+import log from 'electron-log'
+
+const START_PORT = 8000
+const MAX_ATTEMPTS = 10
+
+export const startServer = () => {
+  let attempts = 0
+
+  function attemptAStart() {
+    if (attempts >= MAX_ATTEMPTS) {
+      return Promise.reject(`Failed to bind socket server after ${MAX_ATTEMPTS} attempts.`)
+    }
+
+    return new Promise((resolve, reject) => {
+      const randomPort = START_PORT + Math.floor(1000 * Math.random())
+      log.info(`Starting socket server on port: ${randomPort}`)
+
+      const server = fork(join(__dirname, './socketServer.bundle.js'), [randomPort])
+      server.on('close', (code) => {
```

Comment:
if this happens sometime after the initial startup, it looks like it restarts the server on a new port. Does the rest of the app know that it's now on a new port?

Reply:
No, good point.  The plan was to broadcast the change, but I never got around to it.
I'll create a follow-up PR with this and any other changes.